src/reporter/index.js

```diff
@@ -89,13 +89,13 @@ export default class Reporter {
             reportItem     = this.reportQueue.shift();
             currentFixture = reportItem.fixture;
 
-            await this.plugin.reportTestDone(reportItem.test.name, reportItem.testRunInfo, reportItem.test.meta);
-
             // NOTE: here we assume that tests are sorted by fixture.
             // Therefore, if the next report item has a different
             // fixture, we can report this fixture start.
             nextReportItem = this.reportQueue[0];
 
+            await this.plugin.reportTestDone(reportItem.test.name, reportItem.testRunInfo, reportItem.test.meta);
```

Comment:
At present, we raise the `reportFixtureStart` event after `reportTestDone` event (by some conditions) in the `Reporter` class.

I propose to use an another algorithm.
1)Raise an `fixture-start` event inside the `BrowserJob` class and pass it via `BrowserJob`-> `Task` -> `Reporter` chain.
2)Calculate the first test of each fixture [here](https://github.com/DevExpress/testcafe/blob/master/src/runner/browser-job.js#L25)
3)Raise an `fixture-start` event before emitting the `'test-run-start'` if current test is first test of fixture
https://github.com/DevExpress/testcafe/blob/5b4979d60ced70c54e7e331847c60d90ec8be3cb/src/runner/browser-job.js#L42